cpp/models/IDE/IDE.cpp

```diff
@@ -0,0 +1,168 @@
+#include "IDE/IDE.h"
+#include "memilio/epidemiology/contact_matrix.h"
+#include <iostream>
+
+namespace mio
+{
+
+using Vec = TimeSeries<double>::Vector;
+
+/**
+    * Constructor of IDEModel
+    * @param init TimeSeries with 3d Vectors in its entries containing the initial (time, quantity of susceptible, R0t) values
+    *   the time values should have a distance of dt_init; we need time values from -(k-1)*dt until 0 -> condition: length_init>= k
+    *   (with k=std::ceil((infectiousTime+latencyTime)/dt)) with default values: take 11.6 and you are safe
+    * @param dt_init size of the time step used for numerical integration
+    * @param N population of the considered Region 
+    */
+IdeModel::IdeModel(TimeSeries<double>&& init, double dt_init, int N_init)
+    : result{std::move(init)}
+    , dt{dt_init}
+    , N{N_init}
+{
+    l = (int)std::floor(latencyTime / dt);
+    k = (int)std::ceil((infectiousTime + latencyTime) / dt);
+}
+
+/**
+    * Change latencytime, default value: 3.3
+    * @param latency The value of the new latency time
+    */
+void IdeModel::set_latencytime(double latency)
+{
+    latencyTime = latency;
+    //initialize parameters k and  l used for numerical calculations in the simulation
+    l = (int)std::floor(latencyTime / dt);
+    k = (int)std::ceil((infectiousTime + latencyTime) / dt);
+}
+
+/**
+    * Change infectioustime, default value: 8.2
+    * @param infectious The value of the new infectious time
+    */
+void IdeModel::set_infectioustime(double infectious)
+{
+    infectiousTime = infectious;
+    k              = (int)std::ceil((infectiousTime + latencyTime) / dt);
+}
+
+/**
+    * Get Reference to used ContactMatrix
+    */
+ContactMatrix& IdeModel::get_contact_matrix()
+{
+    return contact_matrix;
+}
+
+/**
+    * Density of the generalized beta distribution used for the function f_{beta} of the IDGL model.
+    * 
+    * @param tau Functionparameter
+    * @param p parameter p of the generalized Beta distribution
+    * @param q parameter q of the generalized Beta distribution
+    */
+double IdeModel::Beta(double tau, double p, double q) const
```

Comment:
better choose longer names if this helps the understanding. I.e. name it generalized_beta_distribution.